cmd/flux/build_kustomization_test.go

```diff
@@ -171,8 +171,7 @@ spec:
 	if err != nil {
 		t.Fatal(err)
 	}
-
```

Comment:
There is more here that could be done, as using `./testdata/` for something temporarily seems wrong.

I think we should instead use `t.TempDir()` as a target for the file that is been written. Which would prevent having to think about garbage collecting it later on.

This is however beyond the time I have available now, as I was merely trying to fix things I ran into while working on #3587.